docs/source/basic/data_prep.md

```diff
@@ -8,11 +8,70 @@ The aim of data preparation is to convert your data to a format that the energon
 
+## Important Considerations
+
+Depending on what your data looks like and how you are planning to use it, you will have to make a few choices,
+**before you prepare your dataset**:
+
+**Monolithic Dataset vs. Polylithic (primary and auxiliary) Datasets**
+
+You can include the media (images/video/audio) inside the same webdataset along with the text and metadata of each sample.
+Or you can keep the media separate (either in another indexed webdataset or as individual files on disk)
+
+If you can, you should go for the monolithic option, because it's faster to load.
+However, there are a few reasons why the other option may be needed:
+
+* You need to keep the original media and you don't want to duplicate it
+* Your media data is very large (e.g. long videos) and you need to keep your primary dataset small
+* You want to re-use the same media with different labels or you want to train on different subsets
+* You want to train with [online packing](../advanced/packing.md) and can't fit all the media of the packing buffer in memory
+
+**How to shard the data**
+
+The WebDataset will be split into a bunch of shards (i.e. tar files). You'll have to decide how many samples to put in one shard and how many shards to get overall.
+
+To maximize the loading speed, use as few shards as possible. Even a single shard can work well!
+However, if you cannot handle files above a certain size you may need to split the shards more.
+A good rule of thumb is to keep your **number of shards below 10k**.
+
+If you are using remote filesystems like S3, there may be an opposing constraint: S3 [limits the number of requests per second](https://docs.aws.amazon.com/AmazonS3/latest/userguide/optimizing-performance.html)
+that you can make for a single prefix (e.g. filename). By using more shards, you can increase the overall rate. Ideally, you would still want to stay below 10k shards.
+
+
+**Raw vs. baked data**
+
+When using images for example, you could put either the encoded JPG, the decoded pixel values or even the encoded features into the dataset.
+
+Typically, we recommend to go with the "original form" (e.g. JPG) and do all the processing on the fly inside the [cooker](crude-data) and [task encoder](../basic/task_encoder).
+This way, you can change the processing and keep your dataset.
+
+However, if the processing becomes a bottleneck, you can move some of it into the dataset creation phase by baking the information in.
+
+Keep in mind that others may also want to use your dataset for a different project.
+
+
+(monolithic-dataset)=
+## Steps to Create a Monolithic Dataset
+
 These are the typical steps to get your data ready:
 
-1. Create a normal [WebDataset](https://github.com/webdataset/webdataset) from your data
+1. Create a normal [WebDataset](https://github.com/webdataset/webdataset) from your data (including all the media content)
 2. Run our preparation tool `energon prepare` to convert to an energon-compatible format
```

Comment:
"convert to an energon-compatible format" sounds like it transforms the webdataset, which we do not. We just create additional files. I think we should mention that.